tools/sdk/lifinity/lifinity.ts

```diff
@@ -22,334 +23,321 @@ export const buildLifinity = ({
 }) => {
   return new Program(
     LifinityAmmIDL,
-    new PublicKey(AMM_PROGRAM_ADDR),
+    AMM_PROGRAM_ADDR,
     new Provider(connection, wallet, Provider.defaultOptions()),
   );
 };
 
-const getWalletNftAccounts = async ({
+export const getWalletNftAccounts = async ({
   connection,
   wallet,
 }: {
   connection: Connection;
   wallet: PublicKey;
 }): Promise<{
-  lifinityTokenAccount: string | undefined;
-  lifinityMetaAccount: string | undefined;
-}> => {
-  let lifinityTokenAccount;
-  let lifinityMetaAccount;
-  const parsedTokenAccounts = await connection.getParsedTokenAccountsByOwner(
+  lifinityNftAccount: PublicKey;
+  lifinityNftMetaAccount: PublicKey;
+} | null> => {
+  const {
+    value: parsedTokenAccounts,
+  } = await connection.getParsedTokenAccountsByOwner(
     wallet,
     {
       programId: TOKEN_PROGRAM_ID,
     },
     'confirmed',
   );
-  for (const tokenAccountInfo of parsedTokenAccounts.value) {
-    const tokenAccountPubkey = tokenAccountInfo.pubkey;
-    const parsedInfo = tokenAccountInfo.account.data.parsed.info;
-    if (parsedInfo.tokenAmount.amount > 0) {
-      const mintAddress = parsedInfo.mint;
-
-      const metadataPDA = await mplTokenMetadata.Metadata.getPDA(mintAddress);
-      const mintAccInfo = await connection.getAccountInfo(metadataPDA);
-      if (mintAccInfo) {
-        const {
-          data: { updateAuthority },
-        } = mplTokenMetadata.Metadata.from(
-          // @ts-ignore
-          new mplCore.Account(mintAddress, mintAccInfo),
-        );
-        if (
-          updateAuthority === 'BihU63mFnjLaBNPXxaDj8WUPBepZqqB4T2RHBJ99f2xo' ||
-          updateAuthority === 'H5q7Z2FJ5KaWmtGquGqoYJYrM73BEpoabzas5y12s38T'
-        ) {
-          lifinityTokenAccount = tokenAccountPubkey;
-          lifinityMetaAccount = metadataPDA;
-          break;
-        }
-      }
+
+  for (const tokenAccountInfo of parsedTokenAccounts) {
+    const {
+      pubkey: tokenAccountPubkey,
+      account: {
+        data: {
+          parsed: {
+            info: {
+              mint,
+              tokenAmount: { amount },
+            },
+          },
+        },
+      },
+    } = tokenAccountInfo;
+
+    if (amount <= 0) {
+      continue;
+    }
+
+    const metadataPDA = await mplTokenMetadata.Metadata.getPDA(mint);
+    const mintAccInfo = await connection.getAccountInfo(metadataPDA);
+
+    if (!mintAccInfo) {
+      continue;
+    }
+
+    const {
+      data: { updateAuthority },
+    } = mplTokenMetadata.Metadata.from(
+      // @ts-ignore
+      new mplCore.Account(mint, mintAccInfo),
+    );
+
+    if (
+      updateAuthority !== 'BihU63mFnjLaBNPXxaDj8WUPBepZqqB4T2RHBJ99f2xo' &&
+      updateAuthority !== 'H5q7Z2FJ5KaWmtGquGqoYJYrM73BEpoabzas5y12s38T'
+    ) {
+      continue;
```

Comment:
isn't this always true? since updateAuthority holds only one value

Reply:
I have no idea why theses lines exists ^^

Reply:
added a comment about the fact I have no idea why this check exists

Reply:
no, if the key is not `BihU63mFnjLaBNPXxaDj8WUPBepZqqB4T2RHBJ99f2xo` or `H5q7Z2FJ5KaWmtGquGqoYJYrM73BEpoabzas5y12s38T` then it **continue** otherwise it does.